### ISSUE TITLE
False positive for SC2154 when dereferencing an associative array inside arithmetic evaluation command

### ISSUE THREAD

**Issue by ngkz** [koalaman/shellcheck#1074]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
declare -A test
(( test[foo] ))
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 3:
(( test[foo] ))
        ^-- SC2154: foo is referenced but not assigned.

$ 
```

#### Here's what I wanted or expected to see:
No output

**Comment by koalaman** [comment]
Well caught. Thanks!
